builder/amazon/ebs/step_create_ami.go

```diff
@@ -52,7 +52,9 @@ func (s *stepCreateAMI) Run(_ context.Context, state multistep.StateBag) multist
 
 	ui.Say("Waiting for AMI to become ready...")
 	if _, err := awscommon.WaitForState(&stateChange); err != nil {
-		err := fmt.Errorf("Error waiting for AMI: %s", err)
+		imagesResp, _ := ec2conn.DescribeImages(&ec2.DescribeImagesInput{ImageIds: []*string{createResp.ImageId}})
```

Comment:
let's check for an error code here. If this fails then I think we'll have a null pointer on the line below. If the wait failed because we got rate limited, then this would fail too and we would have a panic.

Reply:
makes sense: thanks for your comments. I've submitted a more difensive version to check for null pointers.